tests/consensus_tests/test_shard_transfer_deferred.py

```
@@ -292,6 +292,103 @@ def test_shard_transfer_includes_deferred_points(tmp_path: pathlib.Path, transfe
+      consensus apply path -> the apply never returns, the consensus
+      broadcast never fires, POST /cluster times out.
+
+    This test asserts the symptom: POST /cluster returns within a sane
+    deadline. Once the engine bug is fixed (e.g. wait_for_deferred_points_ready
+    polling feedback closure, or the optimizer firing optimization_finished
+    even when skipping), this test will pass without any workarounds in the
+    test code itself.
+    """
+    assert_project_root()
+
+    peer_api_uris, peer_dirs, bootstrap_uri = start_cluster(tmp_path, N_PEERS)
```

Ruff (0.15.12)

[warning] Unpacked variable `peer_dirs` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)